dot/rpc/subscription/websocket.go

```diff
@@ -333,14 +333,15 @@ func (c *WSConn) initExtrinsicWatch(reqID float64, params interface{}) (Listener
 	c.mu.Unlock()
 
 	err = c.CoreAPI.HandleSubmittedExtrinsic(extBytes)
-	if errors.Is(err, runtime.ErrInvalidTransaction) || errors.Is(err, runtime.ErrUnknownTransaction) {
-		c.safeSend(newSubscriptionResponse(authorExtrinsicUpdatesMethod, extSubmitListener.subID, "invalid"))
-		return nil, err
-	} else if err != nil {
-		c.safeSendError(reqID, nil, err.Error())
-		return nil, err
+	if err != nil {
+		var txnValidityErr *runtime.TransactionValidityError
+		if errors.As(err, &txnValidityErr) {
+			c.safeSend(newSubscriptionResponse(authorExtrinsicUpdatesMethod, extSubmitListener.subID, "invalid"))
+		} else {
+			c.safeSendError(reqID, nil, err.Error())
+		}
+		return nil, fmt.Errorf("handling submitted extrinsic: %w", err)
```

Comment:
nit do we want to return an error from this function if the error is a transaction validity error? 🤔 
Might be, I'm just curious

Reply:
I believe so but maybe I'm missing something. It was this way before. Why would we not?

Reply:
I'm just wondering since it's actually being handled with `c.safeSend`.

Reply:
It sends and logs it, but since no error isn't returned I think this is just a way to notify caller of the error

Reply:
Yes let's keep it as is, re-reading my question it was a silly one. We should log errors even though we safe send it to whatever.